### ISSUE TITLE
not an issue

### ISSUE THREAD

**Issue by ploughedfieldman123** [Stephan-S/FS19_AutoDrive#126]
just like to ask how can you use courseplay with AutoDrive

**Comment by fudoido** [comment]
there is a video on youtube teaching how to use unload from combine function, tractor+trailer using autodrive, combine using courseplay (fieldwork mode for harvesting), but everything is still fresh since Stephan and courseplay developer team only started cooperating with each otehr very recently.
You will need both mods latest versions to make it work properly, and keep looking both mods because updates will come very fast to suport this cooperation between autodrive and courseplay.

**Comment by ploughedfieldman123** [comment]
where on that video do it mention about course play.  I know how to unload combines at a field.

**Comment by Anonymous-any** [comment]
Heyyaiiii amazing news ! THAAAAANNNNNKKKKS @Stephan-S @pvaiko for investigate ! look really userfull
Cannot wait :p :+1:

**Comment by ploughedfieldman123** [comment]
> Heyyaiiii amazing news ! THAAAAANNNNNKKKKS @Stephan-S @pvaiko for investigate ! look really userfull
> Cannot wait :p 👍

what are you on about whats  amazing news

**Comment by fudoido** [comment]
@ploughedfieldman123 the video doesn't mention courseplay, but teaches you how to use unload mode on autodrive. Instead of using the regular helper on combine, use courseplay with fidwork mode, autodrive will work the same way to collect full combine.

**Comment by ploughedfieldman123** [comment]
Oh so you mean use course play just to harvest the field, then when combine full make autodrive go to empty combine  each time

**Comment by fudoido** [comment]
Exactly. That is how it's ment to work. Autodrive does not work on fields. Courseplay will do field work, and when combine full, it will call autodrive tractor to unload. When autodrive tractor full, it will drive to silo or selling point and go back to field wait for full combine again.

**Comment by pvaiko** [comment]
@fudoido I must mention here the great pathfinding function of Courseplay which avoids the fruit on the field when CP is driving on the field to unload or refill...

**Comment by Stephan-S** [comment]
@pvaiko to be fair though, AutoDrive also has fruit avoidance in it's pathfinding on the field ;-)
The main advantage of the CP driver is efficiency, since it can unload while driving and can be called in advance.

**Comment by pvaiko** [comment]
Oh, sweet. We gotta do a pathfinding competition one day :)

**Comment by Anonymous-any** [comment]
@ploughedfieldman123 I'm just so happy about learn CP and AD try to do something fusion. I don't want disturb you so.

**Comment by ploughedfieldman123** [comment]
> Exactly. That is how it's ment to work. Autodrive does not work on fields. Courseplay will do field work, and when combine full, it will call autodrive tractor to unload. When autodrive tractor full, it will drive to silo or selling point and go back to field wait for full combine again.
> 
> Dont forget in courseplay options for combine, to make it stop while unloading, because autodrive does not follow discharge pipe from combine.

Well mine isn't working the tractor is in field with trailer. I have combine on field work its now full but tractor wont go to combine to unload it 

[url=https://ibb.co/3vCgn0b][img]https://i.ibb.co/7Q1sqGD/ADCOMBINE.jpg[/img][/url]


[url=https://ibb.co/StKXvSB][img]https://i.ibb.co/bsrgdtX/ADCOMBINE2.jpg[/img][/url]

**Comment by fudoido** [comment]
Try to move camera to other vehicle, out of that area. For some reason I noticed sometimes it only triggers when you re not focusing on the courseplay/autodrive vehicles, when they are out of camera range. Do that, wait a minute, then go back to check if it started.
Can't see a reason why it doesn't trigger.

**Comment by fudoido** [comment]
Also, try to change combine autodrive to the drive mode instead of unload or combine mode. Video I shared above has Combine with drive mode. That is I have set up mine also and works.

**Comment by Anonymous-any** [comment]
Let me know something normally AD HUD ‎should not be set on combine mode in the combine side right? only the field number is required as normal default mode it showed in the video combine side...
@ploughedfieldman123 show in 2nd screen mistake ?

**Comment by ploughedfieldman123** [comment]
> Try to move camera to other vehicle, out of that area. For some reason I noticed sometimes it only triggers when you re not focusing on the courseplay/autodrive vehicles, when they are out of camera range. Do that, wait a minute, then go back to check if it started.
> Can't see a reason why it doesn't trigger.

How do you mean try to move camera out of that area, do mean when at tractor how can I move camera . do you mean when in tractor, look away from combine




> Also, try to change combine autodrive to the drive mode instead of unload or combine mode. Video I shared above has Combine with drive mode. That is I have set up mine also and works.

so put combine autodrive to be on just drive mode but with field 19 showing only do I need to start AD or not so combine is in green for field 19

**Comment by ploughedfieldman123** [comment]
I have done what you said but still dont work

<a href="https://ibb.co/gr3SQHj"><img src="https://i.ibb.co/zmrXCjn/AD1.jpg" alt="AD1" border="0"></a><br /><a target='_blank' href='http://www.statewideinventory.org/ford-0-60-times/'>taurus sho 0 60</a><br />

<a href="https://ibb.co/Yf6JGBr"><img src="https://i.ibb.co/RQ1tWT5/AD2.jpg" alt="AD2" border="0"></a><br /><a target='_blank' href='http://www.statewideinventory.org/ford-0-60-times/'>taurus sho 0 60</a><br />

**Comment by fudoido** [comment]
Select normal drive mods, select field 19 target, but do not start autodrive for combine.
Set tractor with trailer on "combine unload" mode, set field 19 target, start autodrive for tractor (just like it is now). 
Moving camera I mean to go other place in map, away from combine and tractor. I noticed sometimes vehicles don't start if you are looking at them, for some strange reason. By moving to other place of map and coming back later they start working.

**Comment by ploughedfieldman123** [comment]
when you say normal drive mod so arrow is pointing upwards 

its just not working at all  I even got out of tractor 

<a href="https://ibb.co/92fXFTc"><img src="https://i.ibb.co/mtYsLqb/AD3.jpg" alt="AD3" border="0"></a><br /><a target='_blank' href='http://www.statewideinventory.org/ford-0-60-times/'

**Comment by fudoido** [comment]
I mean the button to change modes, drive unload or combine mode. Drive is the icon with arrow on the button, Unload is trailer icon and combine unload is combine icon.
Set combine to show button with arrow icon, select field 19 as target, but do not start autodrive.

**Comment by fudoido** [comment]
Other question, do you have the course (green lines) properly connected?
Have you refreshed autodrive courses before starting autodrive? (the 2 circular arrow button)

**Comment by ploughedfieldman123** [comment]
> Other question, do you have the course (green lines) properly connected?
> Have you refreshed autodrive courses before starting autodrive? (the 2 circular arrow button)



Hi  I deleted a waypoint 1 arrow joined it back up again did the refresh calc routes. it didn't go straight away. but after a few mins it went to unload combine ok.  parked where it unloaded combine. then went back to field 19 start.   next time combine was 100% full it reversed what seem weird. stopped turned off engine, but not pipe comes out. so there-fore tractor on AD didn't go to empty it.  even if i manually extended combine arm
